ontopy/ontology.py

```diff
@@ -343,12 +344,13 @@ def get_by_label(
                 label_annotations=label_annotations,
                 prefix=prefix,
             )
-            if len(entitylist) > 0:
-                return entitylist[0]
+            if len(entitylist) == 1:
+                return next(iter(entitylist))
 
             raise NoSuchLabelError(
-                f"No label annotations matches {label!r} with prefix "
-                f"{prefix!r}"
+                f"Either no label annotations matches for {label!r} "
```

Comment:
Maybe better to separate the two cases with `NoSuchLabelError` (entitylist is empty) and `AmbigiousLabelError` (entitylist has more than one element).